urlparser.js

```diff
@@ -0,0 +1,30 @@
+var urlparse = 
+    function getUrlParam(key, url) {
+        if (url === undefined) {
+            url = window.location.search;
+        }
+        var urlParams = url.split("?")[1];
+        if (urlParams === undefined) {
+            return;
+        }
+        if(urlParams.match(/#/)){ 
+            //Removes anchor/url fragments from the end of the url
+           urlParams = urlParams.slice(0,urlParams.indexOf('#'));
+        }
+        var value = null;
+        urlParams = urlParams.split("&");
+        urlParams.forEach(function (param) {
+            let parameter = {
+                key: param.split("=")[0],
+                value: param.split("=")[1]
+            }
+            if (parameter.key === key) {
+                value = parameter.value || null;
```

Comment:
If the value is `0` this will overwrite it to `null`.

Reply:
That's another good test to add :)

Reply:
Actually, since `?key=0` would end up with a value being the string `"0"`, then this is probably fine.
Write a test first and see if it passes :)

Reply:
Yep, new test returns string "0" which is appropriate, since null should only return if the value isn't specified but the key exists